wallet/src/Ergvein/Wallet/Monad/Auth.hs

```diff
@@ -324,33 +324,56 @@ instance (MonadBaseConstr t m, HasStoreDir m) => MonadStorage t (ErgveinM t m) w
-        . currencyPubStorage'transactions . at txid .~ Just etx
+      ai' <- modifyExternalRef authRef $ \ai ->
+        let ai' = ai
+              & authInfo'storage
+              . storage'pubStorage
+              . pubStorage'currencyPubStorages
+              . at cur . _Just                  -- TODO: Fix this part once there is a way to generate keys. Or signal an impposible situation
+              . currencyPubStorage'transactions . at txid .~ Just etx
+        in (ai', ai')
+      storeWalletPure $ Just ai'
   {-# INLINE addTxToPubStorage #-}
+  addTxMapToPubStorage txmapE = do
+    authRef <- asks env'authRef
+    performFork_ $ ffor txmapE $ \(cur, txm) -> do
+      ai' <- modifyExternalRef authRef $ \ai ->
+        let ai' = ai
+              & authInfo'storage
+              . storage'pubStorage
+              . pubStorage'currencyPubStorages
+              . at cur . _Just
+              . currencyPubStorage'transactions %~ (M.union txm)
+        in (ai', ai')
+
+      storeWalletPure $ Just ai'
+
   getPubStorageD = do
     authInfoD <- externalRefDynamic =<< asks env'authRef
     pure $ ffor authInfoD $ \ai -> ai ^. authInfo'storage. storage'pubStorage
   {-# INLINE getPubStorageD #-}
   setLabelToExtPubKey reqE = do
     authRef <- asks env'authRef
-    performFork_ $ ffor reqE $ \(cur, i, l) -> modifyExternalRefMaybe_ authRef $
-      updateKeyBoxWith cur i $ \kb -> kb {extKeyBox'key = updateKeyLabel l $ extKeyBox'key kb}
+    performFork_ $ ffor reqE $ \(cur, i, l) -> do
+      mai <- modifyExternalRefMaybe authRef $
+        updateKeyBoxWith cur i $ \kb -> kb {extKeyBox'key = updateKeyLabel l $ extKeyBox'key kb}
+      storeWalletPure mai
 
-  setFlatToExtPubKey reqE = do
+  setFlagToExtPubKey reqE = do
     authRef <- asks env'authRef
-    performFork_ $ ffor reqE $ \(cur, i) -> modifyExternalRefMaybe_ authRef $
-      updateKeyBoxWith cur i $ \kb -> kb {extKeyBox'manual = True}
+    performFork_ $ ffor reqE $ \(cur, i) -> do
+      mai <- modifyExternalRefMaybe authRef $
+        updateKeyBoxWith cur i $ \kb -> kb {extKeyBox'manual = True}
+      storeWalletPure mai
 
   insertTxsInPubKeystore reqE = do
     authRef <- asks env'authRef
-    performFork_ $ ffor reqE $ \(cur, i, txids) -> modifyExternalRefMaybe_ authRef $
-      updateKeyBoxWith cur i $ \kb -> kb {extKeyBox'txs = S.union (extKeyBox'txs kb) $ S.fromList txids}
+    performFork_ $ ffor reqE $ \(cur, i, txids) -> do
+      mai <- modifyExternalRefMaybe authRef $
+        updateKeyBoxWith cur i $ \kb -> kb {extKeyBox'txs = S.union (extKeyBox'txs kb) $ S.fromList txids}
+      storeWalletPure mai
 
-updateKeyBoxWith :: Currency -> Int -> (EgvExternalKeyBox -> EgvExternalKeyBox) -> AuthInfo -> Maybe AuthInfo
+updateKeyBoxWith :: Currency -> Int -> (EgvExternalKeyBox -> EgvExternalKeyBox) -> AuthInfo -> Maybe (AuthInfo, AuthInfo)
```

Comment:
The change will break incoming new methods from new scan. And also it introduce additional complexity where it shouldn't be.